### DISCUSSION TITLE
Podman adds DNS entries in resolv.conf inside container when no dns entry in host resolv.conf

### DISCUSSION THREAD

**Discussion by sriram241294** [containers/podman#19037] [Q&A] [answered]
Hi,
I'm having an IPV6 enabled VM where podman is running. There are cases when the host machine does not have any nameserver configured in the resolv.conf. In the case when the host resolv.conf does not have any nameserver configured i can see podman adding the below entries to the container resolv.conf. 

![image](https://github.com/containers/podman/assets/14277167/0b5a019e-6cbb-46e3-9d5b-9c4d87208ef7)

So my question is why is it adding ipv4 default google dns entries in IPV6 machine. My VM has only IPV6 network enabled.

And is there a way to tell podman not to add any entries into the container or add only IPV6 google DNS entries??

**Comment by Luap99** [comment, marked as answer]
If no suitable nameserver is found then podman will default to the google ones. There is extra logic to detect ipv4 only and not add ipv6 servers in such case but there is no such check for ipv6 only so that is definitely something that can be improved.

I am not sure why it would add nameservers when there are already some added. That looks like a bug, it should only add them if no servers are found. What podman version is this?

If you want full control over the dns use the `--dns` option. Or set the dns servers in containers.conf.

**Reply by ravi5802** [reply]
i have tried by uncommenting the dns_options = [] on /usr/share/containers/containers.conf fileand i have ran podman run --dns --name conatinername containerid but geeting blelow error
Error: -d is not an ip address


but no

**Reply by sriram241294** [reply]
Hi @Luap99 ,
Thanks for your answer.
On the host machine there is no dns entries that's the reason its adding google dns here. The screenshot i have is of the container and not the host.
Im using podman 4.2.0 fyi.
